core/src/main/java/org/mule/runtime/core/MessagePropertiesContext.java

```diff
@@ -219,79 +247,62 @@ public String toString()
         return buf.toString();
     }
 
-    /**
-     * Check for properties that can't be serialized
-     */
-    private void writeObject(java.io.ObjectOutputStream out) throws IOException
+    private <T extends Serializable> T getValueOrDefault(T value, T defaultValue)
```

Comment:
There are only two usages of this method, which build the params in a similar way. Can that logic be brought to this method?